AWSPinpoint/AWSPinpointEndpointProfile.m

```diff
@@ -122,7 +122,9 @@ - (BOOL) isApplicationLevelOptOut:(AWSPinpointContext *) context {
 }
 
 - (void) setEndpointOptOut:(BOOL) applicationLevelOptOut {
-    BOOL isOptedOutForRemoteNotifications = ![AWSPinpointNotificationManager isNotificationEnabled];
+    BOOL isNotificationEnabled = [AWSPinpointNotificationManager isNotificationEnabled];
+    BOOL isOptedOutForRemoteNotifications = !(isNotificationEnabled && [_address length]);
+
```

Comment:
1.  We can use self.address.length rather than _address.  Doing so will use the accessor which has the `nonatomic` and `readwrite` modifiers to the accessor.
2.  I believe your logic is correct, but it might be easier to read if we change it to the following:
```
    BOOL isUsingPinpointForNotifications = [AWSPinpointNotificationManager isNotificationEnabled] && self.address.length
    BOOL isOptedOutForNotifications = !isUsingPinpointForNotifications
```
I think this is easier because `isUsingPinpointForNotifications` describes a case where the user has NotificationsEnabled and the developer has actually called the `intercept` call and passed in the deviceid.

Reply:
thanks, updating!